IMPORT LIBRARIES and DATASET


In [52]:
import pandas as pd 
df = pd.read_csv(r'C:\Users\mhdih\OneDrive\Desktop\uni\D3_project_files\movies_metadata.csv', low_memory=False)
df.head()

adult                              belongs_to_collection    budget  \
0  False  {'id': 10194, 'name': 'Toy Story Collection', ...  30000000   
1  False                                                NaN  65000000   
2  False  {'id': 119050, 'name': 'Grumpy Old Men Collect...         0   
3  False                                                NaN  16000000   
4  False  {'id': 96871, 'name': 'Father of the Bride Col...         0   

                                              genres  \
0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
1  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   
2  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...   
3  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   
4                     [{'id': 35, 'name': 'Comedy'}]   

                               homepage     id    imdb_id original_language  \
0  http://toystory.disney.com/toy-story    862  tt0114709                en   
1                                   NaN   8844  tt0113497                en   
2                                   NaN  15602  tt0113228                en   
3                                   NaN  31357  tt0114885                en   
4                                   NaN  11862  tt0113041                en   

                original_title  \
0                    Toy Story   
1                      Jumanji   
2             Grumpier Old Men   
3            Waiting to Exhale   
4  Father of the Bride Part II   

                                            overview  ... release_date  \
0  Led by Woody, Andy's toys live happily in his ...  ...   1995-10-30   
1  When siblings Judy and Peter discover an encha...  ...   1995-12-15   
2  A family wedding reignites the ancient feud be...  ...   1995-12-22   
3  Cheated on, mistreated and stepped on, the wom...  ...   1995-12-22   
4  Just when George Banks has recovered from his ...  ...   1995-02-10   

       revenue runtime                                   spoken_languages  \
0  373554033.0    81.0           [{'iso_639_1': 'en', 'name': 'English'}]   
1  262797249.0   104.0  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...   
2          0.0   101.0           [{'iso_639_1': 'en', 'name': 'English'}]   
3   81452156.0   127.0           [{'iso_639_1': 'en', 'name': 'English'}]   
4   76578911.0   106.0           [{'iso_639_1': 'en', 'name': 'English'}]   

     status                                            tagline  \
0  Released                                                NaN   
1  Released          Roll the dice and unleash the excitement!   
2  Released  Still Yelling. Still Fighting. Still Ready for...   
3  Released  Friends are the people who let you be yourself...   
4  Released  Just When His World Is Back To Normal... He's ...   

                         title  video vote_average vote_count  
0                    Toy Story  False          7.7     5415.0  
1                      Jumanji  False          6.9     2413.0  
2             Grumpier Old Men  False          6.5       92.0  
3            Waiting to Exhale  False          6.1       34.0  
4  Father of the Bride Part II  False          5.7      173.0  

[5 rows x 24 columns]

INFO on Dataset with rows and Datatypes


In [53]:
df.info()
df.shape

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45466 entries, 0 to 45465
Data columns (total 24 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   adult                  45466 non-null  object 
 1   belongs_to_collection  4494 non-null   object 
 2   budget                 45466 non-null  object 
 3   genres                 45466 non-null  object 
 4   homepage               7782 non-null   object 
 5   id                     45466 non-null  object 
 6   imdb_id                45449 non-null  object 
 7   original_language      45455 non-null  object 
 8   original_title         45466 non-null  object 
 9   overview               44512 non-null  object 
 10  popularity             45461 non-null  object 
 11  poster_path            45080 non-null  object 
 12  production_companies   45463 non-null  object 
 13  production_countries   45463 non-null  object 
 14  release_date           45379 non-null  object 
 15  re

(45466, 24)

REMARKS:  
Rows-45466  
Columns-24

Checking for emptiness on the data.This shows the percentage of msising values in each column


In [65]:
# df.isnull().sum()
df.isnull().sum()/(len(df))*100  

adult                     0.000000
belongs_to_collection    90.115691
budget                    0.006598
genres                    0.000000
homepage                 82.883913
id                        0.000000
imdb_id                   0.037391
original_language         0.024194
original_title            0.000000
overview                  2.098271
popularity                0.013197
poster_path               0.848986
production_companies      0.006598
production_countries      0.006598
release_date              0.197950
revenue                   0.013197
runtime                   0.578454
spoken_languages          0.013197
status                    0.191352
tagline                  55.104914
title                     0.013197
video                     0.013197
vote_average              0.013197
vote_count                0.013197
dtype: float64

**ABOVE DATA SHOWS THAT 3 of 24 COLUMNS HAS MORE THAN 50% DATA NULL**  

-BELONGS TO COLLECTION - If the movie not belongs to a collection(Highly possible for most of the data)  
-HOMEPAGE - Homepage of the movie if available/not (Highly unlikely)  
-TAGLINE - No value for us at all  



Summary statistics of the numerical columns


In [66]:
df.describe()

budget    popularity                   release_date  \
count  4.546300e+04  45460.000000                          45376   
mean   4.224579e+06      2.921478  1992-05-15 17:10:50.564174976   
min    0.000000e+00      0.000000            1874-12-09 00:00:00   
25%    0.000000e+00      0.385948            1978-10-06 00:00:00   
50%    0.000000e+00      1.127685            2001-08-30 00:00:00   
75%    0.000000e+00      3.678902            2010-12-17 00:00:00   
max    3.800000e+08    547.488298            2020-12-16 00:00:00   
std    1.742413e+07      6.005414                            NaN   

            revenue       runtime  vote_average    vote_count  
count  4.546000e+04  45203.000000  45460.000000  45460.000000  
mean   1.120935e+07     94.128199      5.618207    109.897338  
min    0.000000e+00      0.000000      0.000000      0.000000  
25%    0.000000e+00     85.000000      5.000000      3.000000  
50%    0.000000e+00     95.000000      6.000000     10.000000  
75%    0.000000e+00    107.000000      6.800000     34.000000  
max    2.787965e+09   1256.000000     10.000000  14075.000000  
std    6.433225e+07     38.407810      1.924216    491.310374

**BUDGET**  
min = 0, means low budget movies/ data entry error  
mean > max, means 
  
**POPULARITY**  
min = 0, Less popular/ data entry error  
mean < max  

**RELEASE DATE**  
1874-12-09 to 2020-12-16 
  
**REVENUE**  
max = 2.79b, Likely a blockbuster film  
mean = 11.2m, Moderate revenue only  

**RUNTIME**  
min = 0, Data entry error probably  
max = 1256 min, Likely a data entry error as its too long!  

**VOTE AVERAGE**  
min = 0, Not votes/ratings probably  
mean = 5.62, slightly above average  

**VOTE COUNT**  
max = 14,075, Highly popular  
mean = 109.90, Means most movies are rated my small number of movies only.  

In [86]:
df.describe(include='object')

adult                              belongs_to_collection  \
count   45466                                               4494   
unique      5                                               1698   
top     False  {'id': 415931, 'name': 'The Bowery Boys', 'pos...   
freq    45454                                                 29   

                               genres                     homepage      id  \
count                           45466                         7782   45466   
unique                           4069                         7673   45436   
top     [{'id': 18, 'name': 'Drama'}]  http://www.georgecarlin.com  141971   
freq                             5000                           12       3   

          imdb_id original_language       original_title            overview  \
count       45449             45455                45466               44512   
unique      45417                92                43373               44307   
top     tt1180333                en  Alice in Wonderland  No overview found.   
freq            3             32269                    8                 133   

                             poster_path production_companies  \
count                              45080                45463   
unique                             45024                22708   
top     /5D7UBSEgdyONE6Lql6xS7s6OLcW.jpg                   []   
freq                                   5                11875   

                                     production_countries  \
count                                               45463   
unique                                               2393   
top     [{'iso_3166_1': 'US', 'name': 'United States o...   
freq                                                17851   

                                spoken_languages    status  \
count                                      45460     45379   
unique                                      1931         6   
top     [{'iso_639_1': 'en', 'name': 'English'}]  Released   
freq                                       22395     45014   

                       tagline       title  video  
count                    20412       45460  45460  
unique                   20283       42277      2  
top     Based on a true story.  Cinderella  False  
freq                         7          11  45367

REMARKS:  
**ADULT**  
unique - 5 (Other values than TRUE / FALSE available)
frequency - 45454 False values(Majority Non adult movies)
  
**BELONGS TO COLLECTION**  
frequency - The Bowery Boys appears 29 times  

**GENRES**  
frequency -5000 Drama movies in total  

**HOMEPAGE**  
frequency -12 movies from http://www.georgecarlin.com  

**ORIGINAL LANGUAGE**  
32268 en movies available of 92 languages   

**ORIGINAL TITLE**  
frequency - 8 Movies from Alice in Wonderland   

**OVERVIEW**  
No overview in 133 movies  

**POSTER PATH**  
5 movies with the same poster available  

**PRODUCTION COUNTRIES**  
US movies are most available

**SPOKEN LANGUAGES**  
English movies mostly 

**STATUS**  
6 unique values other than RELEASED / RUMOURED  

**TITLE**  
11 movies of Cinderella  

Lets check for **MISSING VALUES** now

In [146]:
print(df['adult'].unique())

['False' 'True' ' - Written by Ørnås'
 ' Rune Balot goes to a casino connected to the October corporation to try to wrap up her case once and for all.'
 ' Avalanche Sharks tells the story of a bikini contest that turns into a horrifying affair when it is hit by a shark avalanche.']


In [157]:
problematic_rows = df[df['adult'].isin([' - Written by Ørnås', 'Rune Balot goes to a casino connected to the October corporation to try to wrap up her case once and for all.', 'Avalanche Sharks tells the story of a bikini contest that turns into a horrifying affair when it is hit by a shark avalanche.'])]
print(problematic_rows)


                     adult belongs_to_collection  budget  \
19730   - Written by Ørnås              0.065736     NaN   

                                                  genres  \
19730  [{'name': 'Carousel Productions', 'id': 11176}...   

                                                homepage          id imdb_id  \
19730  [{'iso_3166_1': 'CA', 'name': 'Canada'}, {'iso...  1997-08-20       0   

      original_language                            original_title  overview  \
19730             104.0  [{'iso_639_1': 'en', 'name': 'English'}]  Released   

       ...  release_date revenue runtime spoken_languages status  tagline  \
19730  ...           NaT     NaN     NaN              NaN    NaN      NaN   

       title video vote_average vote_count  
19730    NaN   NaN          NaN        NaN  

[1 rows x 24 columns]


Checking for uniqueness in values.Help know types and possibilities


In [158]:
df.nunique()

adult                        5
belongs_to_collection     1698
budget                    1223
genres                    4069
homepage                  7673
id                       45436
imdb_id                  45417
original_language           92
original_title           43373
overview                 44307
popularity               43757
poster_path              45024
production_companies     22708
production_countries      2393
release_date             17333
revenue                   6863
runtime                    353
spoken_languages          1931
status                       6
tagline                  20283
title                    42277
video                        2
vote_average                92
vote_count                1820
dtype: int64

NOW FOR 24 COLUMNS, the ones that seem less usable are:  
ADULT -
HOMEPAGE -
OVERVIEW -
POSTER PATH -
VIDEO -
TAGLINE -


In [159]:
df_update = df.drop(columns=['adult', 'homepage', 'overview', 'poster_path', 'video', 'tagline'])  
df_update.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45466 entries, 0 to 45465
Data columns (total 18 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   belongs_to_collection  4494 non-null   object        
 1   budget                 45463 non-null  float64       
 2   genres                 45466 non-null  object        
 3   id                     45466 non-null  object        
 4   imdb_id                45449 non-null  object        
 5   original_language      45455 non-null  object        
 6   original_title         45466 non-null  object        
 7   popularity             45460 non-null  float64       
 8   production_companies   45463 non-null  object        
 9   production_countries   45463 non-null  object        
 10  release_date           45376 non-null  datetime64[ns]
 11  revenue                45460 non-null  float64       
 12  runtime                45203 non-null  float64       
 13  s

In [160]:
df['release_date'] = pd.to_datetime(df['release_date'], errors='coerce')    #The errors='coerce' => NaN
cutoff_date = pd.Timestamp('1900-01-01')    # Define the cutoff date
df_date = df[df['release_date'] >= cutoff_date] # Filter the DataFrame
df_date['release_date']   #List of Removable values


0       1995-10-30
1       1995-12-15
2       1995-12-22
3       1995-12-22
4       1995-02-10
           ...    
45460   1991-05-13
45462   2011-11-17
45463   2003-08-01
45464   1917-10-21
45465   2017-06-09
Name: release_date, Length: 45295, dtype: datetime64[ns]

In [161]:
df['budget'] = pd.to_numeric(df['budget'], errors='coerce')  
df['popularity'] = pd.to_numeric(df['popularity'], errors='coerce')  
print(df['budget'].isnull().sum())    
print(df['popularity'].isnull().sum())    

3
6


In [162]:
df_rumoured = df[df['status'] == "Rumored"]
print(df['status'].isnull().sum())    
print(df_rumoured['status'].isnull().sum())    
print(df_rumoured.info())    
# print(df_rumoured)    

87
0
<class 'pandas.core.frame.DataFrame'>
Index: 230 entries, 682 to 45289
Data columns (total 24 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   adult                  230 non-null    object        
 1   belongs_to_collection  9 non-null      object        
 2   budget                 230 non-null    float64       
 3   genres                 230 non-null    object        
 4   homepage               17 non-null     object        
 5   id                     230 non-null    object        
 6   imdb_id                228 non-null    object        
 7   original_language      229 non-null    object        
 8   original_title         230 non-null    object        
 9   overview               217 non-null    object        
 10  popularity             230 non-null    float64       
 11  poster_path            229 non-null    object        
 12  production_companies   230 non-null    object        
 13  p

In [163]:
rows_with_nan_date = df[df['release_date'].isna()]  
rows_with_nan_date['release_date']


711     NaT
734     NaT
3460    NaT
3628    NaT
5879    NaT
         ..
45148   NaT
45203   NaT
45338   NaT
45410   NaT
45461   NaT
Name: release_date, Length: 90, dtype: datetime64[ns]

In [164]:
rows_with_nan_budget = df[df['budget'].isna()]  
rows_with_nan_budget['budget']


19730   NaN
29503   NaN
35587   NaN
Name: budget, dtype: float64

In [165]:
rows_with_nan_pop = df[df['popularity'].isna()] 
rows_with_nan_pop['popularity']

19729   NaN
19730   NaN
29502   NaN
29503   NaN
35586   NaN
35587   NaN
Name: popularity, dtype: float64

Budget > 0


In [166]:
df['budget'] = pd.to_numeric(df['budget'], errors='coerce')  
df_posbudget = df[df['budget'] > 0]          #ONLY 8890 available >0
# print(df_posbudget['budget'])